cassandra/src/main/java/org/hibernate/ogm/datastore/cassandra/CassandraDialect.java

```diff
@@ -468,4 +475,38 @@ public boolean supportsSequences() {
 	public DuplicateInsertPreventionStrategy getDuplicateInsertPreventionStrategy(EntityKeyMetadata entityKeyMetadata) {
 		return null;
 	}
+
+	@Override
+	public ClosableIterator<Tuple> executeBackendQuery(
+			BackendQuery<String> query, QueryParameters queryParameters) {
+
+		ResultSet resultSet = bindAndExecute(
+				queryParameters.getPositionalParameterValues(),
+				session.newSimpleStatement( query.getQuery() )
+		);
+
+		int first = 0;
+		if ( queryParameters.getRowSelection().getFirstRow() != null ) {
+			first = queryParameters.getRowSelection().getFirstRow();
+		}
+
+		int max = Integer.MAX_VALUE;
+		if ( queryParameters.getRowSelection().getMaxRows() != null ) {
+			max = queryParameters.getRowSelection().getMaxRows();
+		}
+
+		return new ResultSetTupleIterator( resultSet, first, max );
+	}
+
+	@Override
+	public ParameterMetadataBuilder getParameterMetadataBuilder() {
+		return new CassandraParameterMetadataBuilder( session, provider.getMetaDataCache() );
+	}
+
+	@Override
+	public String parseNativeQuery(String nativeQuery) {
+		// we defer the work, since at this point the table may not yet exist in the db and without
+		// the db server supplied metadata or parsing assistance we can't do much meaningful validation.
+		return nativeQuery;
```

Comment:
That's interesting, it indicates that something is wrong with the timing of schema initialization.

Simply deferring renders start-up validation of named queries moot, rather I think we should fix that ordering issue. I've filed [OGM-955](https://hibernate.atlassian.net/browse/OGM-955) and will look into it.